Jenkinsfile

```diff
@@ -0,0 +1,59 @@
+pipeline {
+    agent none
+    environment {
+        MODULE_NAME = "<%= name%>"
+        BUILD_TRIGGER_EXCLUDES = "^.jenkins/"
+    }
+    options {
+        disableResume()
+    }
+    stages {
+        stage('Build') {
+            agent { label 'build' }
+            steps {
+                script {
+                    // only continue build if changes are relevant to this module.
+                    // WARNING: it only checks the last commit!!!
+                    def filesInThisCommitAsString = sh(script:"git diff --name-only HEAD~1..HEAD | grep -v '$BUILD_TRIGGER_EXCLUDES' || echo -n ''", returnStatus: false, returnStdout: true).trim()
+                    def hasChangesInPath = (filesInThisCommitAsString.length() > 0)
+                    echo "${filesInThisCommitAsString}"
+                    if (!currentBuild.rawBuild.getCauses()[0].toString().contains('UserIdCause') && !hasChangesInPath){
+                        currentBuild.rawBuild.delete()
+                        error("No changes detected in the module path for ${MODULE_NAME} (${BUILD_TRIGGER_EXCLUDES})")
+                    }
+                }
+                echo "Aborting all running jobs for ${MODULE_NAME}"
+                script {
+                    abortAllPreviousBuildInProgress(currentBuild)
+                }
+                echo "Building ..."
+                sh "curl -X POST -H \"Content-Type: application/json\" https://chat.pathfinder.gov.bc.ca/hooks/7A8RwSpZPDAqc6uXS/z8nRXX43oifYWELkwJKeLj4e86rtyfvtJdqdvjSE3XoK2sri -d '{\"text\":\"Building wps-web for PR-${CHANGE_ID}.\"}'"
```

Comment:
Please limit Rocket.Chat notifications to TEST and PROD deployments.  Possibly only PROD for builds, if not less.